### PR TITLE
[4995] Add iQTS route

### PR DESCRIPTION
**Pull request by AbigailMcP**
### Context

https://trello.com/c/mMwhWwkT/5043-l-add-support-for-iqts-add-new-route

### Changes proposed in this pull request

- Add new iQTS enum option to training_routes on trainee
- Add new feature flag to ensure this is off in prod (enabled in review, dev and QA)
- Add new `requires_funding?` method to `TrainingRouteManger` which returns true for all routes except for iQTS
- Use `requires_funding?` method to hide funding sections and prevent funding validation for iQTS trainees.

### Guidance to review

- Create an iQTS trainee
- Check that the funding section / details don't appear anywhere
- Check that you can register them, award them, withdraw them etc

### Important business

- [ ] ~Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?~
- [x] Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

File: app/helpers/training_routes_helper.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module TrainingRoutesHelper
```
Comment (AbigailMcP) [inline app/helpers/training_routes_helper.rb:3]: Refactor - taking logic out of view now it's getting a bit more complicated re different hints.

File: app/views/trainees/training_routes/_form_fields.html.erb [hunk 2]
```diff
@@ -7,15 +7,13 @@
   <% else %>
     <%= f.govuk_radio_button :training_route, TRAINING_ROUTE_ENUMS[:assessment_only], label: { text: t("activerecord.attributes.trainee.training_routes.assessment_only") }, link_errors: true %>
 
-    <% TRAINING_ROUTE_FEATURE_FLAGS.each do |training_route| %>
-      <% if FeatureService.enabled?("routes.#{training_route}") %>
-        <%= f.govuk_radio_button(
-          :training_route,
-          TRAINING_ROUTE_ENUMS[training_route],
-          hint: { text: (TRAINING_ROUTE_ENUMS[training_route].starts_with?("provider_led") && t(".provider_led_hint")) },
```
Comment (AbigailMcP) [inline app/views/trainees/training_routes/_form_fields.html.erb:15]: Moved into helper to take this logic out of the view now we have multiple hints for multiple routes.

**Comment by AbigailMcP** [comment]
[[L] Add support for iQTS - add new route](https://trello.com/c/mMwhWwkT/5043-l-add-support-for-iqts-add-new-route)

**Review by markhallen — APPROVED** [review]
LGTM

**Review by edwardhorsford — CHANGES_REQUESTED** [review]
I've had a look with @johndoates - looking good!

* Can the route be alphabetical in the route list
* In the route switcher in the draft, we need `(iQTS)` to be correctly cased:
Screenshot 2022-12-07 at 17 07 01

* Hint for autocomplete needs a full stop.
* Needs a validation message for autocomplete - `Enter the country or territory the trainee is studying in`
* Needs a validation message when country not recognised `Enter the country or territory the trainee is studying in`
* @claire-hughez do you think we should error if they pick `United Kingdom`. Alternately we could remove it from the list. We might want to check with the iQTS team.
* Please make the autocomplete narrower.
* Filters: please put route alphabetical in list
* Seeds for iQTS trainees need a training country

**Comment by AbigailMcP** [comment]
> * Can the route be alphabetical in the route list

done

> * In the route switcher in the draft, we need `(iQTS)` to be correctly cased:
> 
> Screenshot 2022-12-07 at 17 07 01

done, also fixed for HPITT route (which was rendering as "hpitt")

> * Hint for autocomplete needs a full stop.

done

> * Needs a validation message for autocomplete - `Enter the country or territory the trainee is studying in`

done

> * Needs a validation message when country not recognised `Enter the country or territory the trainee is studying in`
> * @claire-hughez do you think we should error if they pick `United Kingdom`. Alternately we could remove it from the list. We might want to check with the iQTS team.

waiting on iQTS team response

> * Please make the autocomplete narrower.

done

> * Filters: please put route alphabetical in list

done

> * Seeds for iQTS trainees need a training country

TBC

**Review by edwardhorsford — APPROVED** [review]
Looking great!

**Pull request merged**